### PR TITLE
feat: implement chart component

### PR DESCRIPTION
**Pull request by AlejandroYanes**

### CODE REVIEW

**Comment by AlejandroYanes** [comment]
fixes #347

File: src/components/Chart/index.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            ),
+        });
+    }
+
+    render() {
+        const { children } = this.props;
+        return (
+            <div>
+                <span className="rainbow-chart_dataset-container" aria-hidden>
+                    <Provider value={{ privateRegister: this.registerDataset }}>
```
Comment (reiniergs) [inline src/components/Chart/index.js]: every time you do something like prop={{ value  }} ... the object gets create in every execution of the render method. You can cache this instead by doing 
```
this.contextValue = {
    privateRegister: this.registerDataset.bind(this),
}
```

then in the component render you pass the contextValue directly .. created only one time

Comment (reiniergs) [reply src/components/Chart/index.js]: other observation, use Symbol to hide private methods and variables

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "react-rainbow-components",
-    "version": "0.6.0",
+    "version": "0.5.1",
```
Comment (reiniergs) [inline package.json]: this is incorrect. Are we going back in time? :)

Comment (AlejandroYanes) [reply package.json]: sorry a got mixed on the merge

**Comment by reiniergs** [comment]
can we implement this without React context, since inside of <Chart /> component you can pass <DataSet /> We can just iterate for the dataset in componentDidUpdate ... get the data passed then update the charts

this implementation does not guarantee reactivity since the charts only is created and updated one time. Every time props values change in Chart or DataSet the chart must be updated.

## tips:
- the charts instance doesn't need to be created multiple times check: https://www.chartjs.org/docs/latest/developers/updates.html

**Pull request merged**